src/Brush.elm

```diff
@@ -0,0 +1,1102 @@
+                                if not model.x then
+                                    model.extent.left
+
+                                else
+                                    px
+                            , bottom =
+                                if not model.y then
+                                    model.extent.bottom
+
+                                else
+                                    py
+                            , right =
+                                if not model.x then
+                                    model.extent.right
+
+                                else
+                                    px
+                            }
+
+                        else
+                            Maybe.withDefault model.extent model.selection
+                in
+                { model
+                    | drag =
+                        Just
+                            { current = ( px, py )
+                            , origin = ( px, py )
+                            , mode = mode
+                            , lock = NoLock
+                            , initialSelection = selection
+                            , element = element
+                            }
+                    , shifting = shifting
+                    , selection = Just selection
+                }
+
+            MouseMove position_ ->
+                hasMoved
+                    { model
+                        | drag =
+                            Maybe.map
+                                (\drag ->
+                                    { drag
+                                        | current = position_
+                                        , lock =
+                                            if drag.lock == NoLock && model.shifting then
+                                                if abs (Tuple.first position_ - Tuple.first drag.current) > abs (Tuple.second position_ - Tuple.second drag.current) then
+                                                    LockY
+
+                                                else
+                                                    LockX
+
+                                            else
+                                                drag.lock
+                                    }
+                                )
+                                model.drag
+                    }
+
+            MouseUp ->
+                { model
+                    | drag = Nothing
+                    , selection =
+                        Maybe.andThen
+                            (\selection ->
+                                if selection.top == selection.bottom || selection.right == selection.left then
+                                    Nothing
+
+                                else
+                                    Just selection
+                            )
+                            model.selection
+                }
+
+            ShiftDown ->
```

Comment:
Should this be `Meta` down?

Reply:
No, meta is only used on the mousedown event, and it's only function is to make it look like you clicked on the overlay, so we don't need a specific key down handler for those.